model/registry/security-rule.yaml

```diff
@@ -0,0 +1,68 @@
+groups:
+  - id: registry.security_rule
+    prefix: security_rule
+    display_name: Security Rule
+    type: attribute_group
+    brief: >
+      Describes security rule attributes. Rule fields are used to capture the specifics of any observer or agent rules
+      that generate alerts or other notable events.
+    attributes:
+      - id: category
+        type: string
+        stability: experimental
+        brief: >
+          A categorization value keyword used by the entity using the rule for detection of this event
+        examples: ['Attempted Information Leak']
+      - id: description
+        type: string
+        stability: experimental
+        brief: >
+          The description of the rule generating the event.
+        examples: ['Block requests to public DNS over HTTPS / TLS protocols']
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          A rule ID that is unique within the scope of an agent, observer,
+          or other entity using the rule for detection of this event.
+        examples: ['101']
+      - id: license
+        type: string
+        stability: experimental
+        brief: >
+          Name of the license under which the rule used to generate this event is made available.
+        examples: ['Apache 2.0']
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the rule or signature generating the event.
+        examples: ['BLOCK_DNS_over_TLS']
+      - id: reference
+        type: string
+        stability: experimental
+        brief: >
+          Reference URL to additional information about the rule used to generate this event.
+        note: >
+          The URL can point to the vendor’s documentation about the rule.
+          If that’s not available, it can also be a link to a more general page describing this type of alert.
+        examples: ['https://en.wikipedia.org/wiki/DNS_over_TLS']
+      - id: ruleset
```

Comment:
since ruleset can have multiple properties beyond name (e.g. id), it could make sense to call it `security_rule.ruleset.name` (or `security_rule.set.name`)

Reply:
also is it common to have ruleset? Is it reasonable to assume that all rules in the set have the same version & license? 
Would they be properties of the rule or ruleset the rule belongs to?

Reply:
As we don't have a usecase for `ruleset.id` I don't see the reason to create it. 

It is common to pack rules into sets, but this doesn't mean that rules are having the same version/license etc. Some rules might change at different points of time while still being a part of the same set.